cherab/core/atomic/gaunt.pyx

```diff
@@ -0,0 +1,152 @@
+# Copyright 2016-2022 Euratom
+# Copyright 2016-2022 United Kingdom Atomic Energy Authority
+# Copyright 2016-2022 Centro de Investigaciones Energéticas, Medioambientales y Tecnológicas
+#
+# Licensed under the EUPL, Version 1.1 or – as soon they will be approved by the
+# European Commission - subsequent versions of the EUPL (the "Licence");
+# You may not use this work except in compliance with the Licence.
+# You may obtain a copy of the Licence at:
+#
+# https://joinup.ec.europa.eu/software/page/eupl5
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the Licence for the specific language governing permissions and limitations
+# under the Licence.
+
+from os import path
+import numpy as np
+import json
+
+from libc.math cimport log10, log, M_PI, sqrt
+from raysect.core.math.function.float cimport Interpolator2DArray
+from cherab.core.utility.constants cimport RYDBERG_CONSTANT_EV, SPEED_OF_LIGHT, ELEMENTARY_CHARGE, PLANCK_CONSTANT
+
+cimport cython
+
+
+DEF EULER_GAMMA = 0.5772156649015329
+
+cdef double PH_TO_EV_FACTOR = PLANCK_CONSTANT * SPEED_OF_LIGHT * 1e9 / ELEMENTARY_CHARGE
+
+
+cdef class FreeFreeGauntFactor():
+    """
+    The base class for temperature-averaged free-free Gaunt factors.
+    """
+
+    cpdef double evaluate(self, double zeff, double temperature, double wavelength) except? -1e999:
+        """
+        Returns the temperature-averaged free-free Gaunt factor for the supplied parameters.
+
+        :param double zeff: Effective Z of the plasma.
+        :param double temperature: Electron temperature in eV.
+        :param double wavelength: Spectral wavelength.
+
+        :return: free-free Gaunt factor
+        """
+        raise NotImplementedError("The evaluate() virtual method must be implemented.")
+
+    def __call__(self, double zeff, double temperature, double wavelength):
+        """
+        Returns self.evaluate(zeff, temperature, wavelength).
```

Comment:
Just use the docstring of `evaluate` here

Reply:
Done.